### PR TITLE
Bokeh optimizations

### PR DESCRIPTION
**Pull request by philippjfr**
This PR is meant to provide various optimizations for the bokeh backend. In particular we have the static data optimization in the bokeh backend preventing the data source from being updated and sent to the frontend. However since styling options may still change, the get_data method still has to be executed to modify color mappers and other style-data mappings. However certain data columns are guaranteed to never change and therefore can be skipped. This can provide significant improvements particularly for large paths, and geoviews geometries.

### CODE REVIEW

File: holoviews/plotting/bokeh/path.py [hunk 1]
```diff
@@ -87,13 +87,14 @@ def get_batched_data(self, element, ranges=None, empty=False):
...
 
             # Apply static styles
-            nvals = len(list(eldata.values())[0])
-            style = styles[zorder]
-            sdata, smapping = expand_batched_style(style, self._batched_style_opts,
-                                                   elmapping, nvals)
-            elmapping.update({k: v for k, v in smapping.items() if k not in elmapping})
-            for k, v in sdata.items():
-                data[k].extend(list(v))
+            if eldata:
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/path.py]: A short comment to briefly explain is being optimized might be helpful here.

**Comment by jlstevens** [comment]
Other than one comment above, it looks fine to me and the tests are green. Is this PR ready to review/merge?

**Comment by philippjfr** [comment]
I think I'm going to go through and look for other opportunities for optimization, so not quite yet.

**Comment by philippjfr** [comment]
Might actually take this opportunity to remove one of the older optimizations which was the ``empty`` to ``BokehPlot.get_data`` methods.

**Comment by philippjfr** [comment]
@jlstevens Ready to merge once tests pass.

**Comment by jlstevens** [comment]
Tests are green. Merging.

**Pull request merged**